src/test/java/com/eppo/sdk/EppoClientTest.java

```diff
@@ -40,654 +39,246 @@
-    pumaAttributes.put("brand_affinity", EppoValue.valueOf(1.00));
-    pumaAttributes.put("purchased_last_30_days", EppoValue.valueOf(false));
-    pumaAttributes.put("loyalty_tier", EppoValue.valueOf("bronze"));
-    actionAttributes.put("puma", pumaAttributes);
-
-    // Get our assigned action
-    Optional<String> stringAssignment =
-        EppoClient.getInstance()
-            .getBanditAssignment(
-                "subject30", "banner-bandit-experiment", subjectAttributes, actionAttributes);
+  public void testReinitializeWithoutForcing() {
+    EppoClient firstInstance = initClient(DUMMY_FLAG_API_KEY);
+    EppoClient secondInstance = new EppoClient.Builder().apiKey(DUMMY_FLAG_API_KEY).buildAndInit();
 
-    // Verify assignment
-    assertTrue(stringAssignment.isPresent());
-    assertEquals("adidas", stringAssignment.get());
-    ArgumentCaptor<BanditLogData> banditLogCaptor = ArgumentCaptor.forClass(BanditLogData.class);
-    verify(mockBanditLogger, times(1)).logBanditAction(banditLogCaptor.capture());
-    BanditLogData capturedBanditLog = banditLogCaptor.getValue();
-    assertEquals(0.8043, capturedBanditLog.getActionProbability(), 0.0002);
+    assertSame(firstInstance, secondInstance);
   }
 
   @Test
-  public void testBanditModelActionAssignmentNoContext() {
-    EppoAttributes subjectAttributes = new EppoAttributes();
-    Set<String> actions = Stream.of("nike", "adidas", "puma").collect(Collectors.toSet());
-
-    // Get our assigned action
-    Optional<String> stringAssignment =
-        EppoClient.getInstance()
-            .getBanditAssignment(
-                "subject39", "banner-bandit-experiment", subjectAttributes, actions);
+  public void testReinitializeWitForcing() {
+    EppoClient firstInstance = initClient(DUMMY_FLAG_API_KEY);
+    EppoClient secondInstance =
+        new EppoClient.Builder().apiKey(DUMMY_FLAG_API_KEY).forceReinitialize(true).buildAndInit();
 
-    // Verify assignment
-    assertTrue(stringAssignment.isPresent());
-    assertEquals("puma", stringAssignment.get());
-    ArgumentCaptor<BanditLogData> banditLogCaptor = ArgumentCaptor.forClass(BanditLogData.class);
-    verify(mockBanditLogger, times(1)).logBanditAction(banditLogCaptor.capture());
-    BanditLogData capturedBanditLog = banditLogCaptor.getValue();
-    assertEquals(0.1613, capturedBanditLog.getActionProbability(), 0.0002);
+    assertNotSame(firstInstance, secondInstance);
   }
 
   @Test
-  public void testBanditControlAction() {
+  public void testPolling() {
+    EppoHttpClient httpClient = new EppoHttpClient(TEST_HOST, DUMMY_FLAG_API_KEY, "java", "3.0.0");
+    EppoHttpClient httpClientSpy = spy(httpClient);
+    TestUtils.setBaseClientHttpClientOverrideField(httpClientSpy);
 
-    Set<String> banditActions =
-        Stream.of("option1", "option2", "option3").collect(Collectors.toSet());
+    new EppoClient.Builder()
+        .apiKey(DUMMY_FLAG_API_KEY)
+        .pollingIntervalMs(20)
```

Comment:
With Java, it was much easier to just use very small timeouts than to mock time out (or mock the Timer) like we do in JavaScript